src/lib/ColorSelect.svelte

```diff
@@ -1,22 +1,51 @@
 <script lang="ts">
-	import { okhsv_to_srgb, srgb_to_okhsv } from './colorconversion'
-	import { eps, picker_size, slider_width, border_size, gap_size } from './constants'
-	import { render_main_image, render_slider_image } from './render'
+	import type { Okhsv, Okhsl, Oklab, Oklch, Rgb } from 'culori'
+	import {
+		useMode,
+		modeRgb,
+		modeOklab,
+		modeOklch,
+		modeOkhsl,
+		modeOkhsv,
+	} from 'culori/fn'
 	import { createEventDispatcher } from 'svelte'
+	import {
+		eps,
+		picker_size,
+		slider_width,
+		border_size,
+		gap_size,
+	} from './constants'
+	import { render_main_image, render_slider_image } from './render'
 
-	export let r = 0
-	export let g = 0
-	export let b = 0
+	export let color: Okhsl | Okhsv | Oklab | Oklch | Rgb
 
-	export let transform = (rgb: number[]) => rgb
+	export let transform = (rgb: Rgb) => rgb
 
 	const dispatch = createEventDispatcher()
 	const width = picker_size + slider_width + gap_size + border_size * 2
 	const height = picker_size + border_size * 2
 
-	$: okhsv = srgb_to_okhsv(r, g, b)
+	const toOkhsl = useMode(modeOkhsl)
+	const toOkhsv = useMode(modeOkhsv)
+	const toOklab = useMode(modeOklab)
+	const toOklch = useMode(modeOklch)
+	const toRgb = useMode(modeRgb)
+
+	$: okhsv = toOkhsv(color)!
 	$: uihsv = scale_to_ui(okhsv)
-	$: update_square(okhsv[0])
+	$: transform_fn = (
+		color.mode === 'okhsl'
+			? toOkhsl
```

Comment:
Nested ternaries are gross (at least to me) but I sometimes struggle with reactive one-liners like this (I suppose it could be an external function, but probably most minifiers will compile it down to code that looks like this anyway, so :shrug:)